pilopt/src/lib.rs

```diff
@@ -427,6 +435,86 @@ fn simplify_expression_single<T: FieldElement>(e: &mut AlgebraicExpression<T>) {
     }
 }
 
+/// Uses associative properties to simplify expressions by regrouping constant terms.
+fn try_simplify_associative_operation<T: FieldElement>(
+    left: &mut AlgebraicExpression<T>,
+    right: &mut AlgebraicExpression<T>,
+    op: AlgebraicBinaryOperator,
+) -> Option<AlgebraicExpression<T>> {
+    if op != AlgebraicBinaryOperator::Add {
+        return None;
+    }
+
+    // Find binary operation and other expression, handling both orderings:
+    // (X + C1) + Other
```

Comment:
I find the use of `C1` here a bit confusing, since it sounds like it's a constant but at this point it is not. Maybe `Y`?

Reply:
Improved the comments to make them clearer and to better follow actual variables in the code. Thanks!